hls4ml/backends/vivado/vivado_backend.py

```diff
@@ -108,6 +108,7 @@ def _register_flows(self):
             'vivado:inplace_parallel_reshape',
             'vivado:inplace_stream_flatten',
             'vivado:skip_softmax',
+            'infer_precision_types',
```

Comment:
As in the quartus case, shouldn't this have vivado: prepended?

Reply:
No, as a general optimizer that doesn't depend on any backend, it needs no prefix. It is used here since backend-specific optimizers can introduce new layers with types that are "auto", so it needs to run again. Similar to how you have to repeat specific_types optimizer in different flows